portal-api/src/v1/controllers/utilisation-report-service/fee-record-corrections/get-fee-record-correction.controller/index.test.ts

```
@@ -0,0 +1,133 @@
+
+    let req: GetFeeRecordCorrectionRequest;
+    let res: MockResponse<Response>;
+
+    beforeEach(() => {
+      req = httpMocks.createRequest<GetFeeRecordCorrectionRequest>({
+        params: aValidRequestQuery(),
+      });
+      res = httpMocks.createResponse();
+
+      jest.mocked(api.getFeeRecordCorrectionById).mockResolvedValue(aGetFeeRecordCorrectionResponseBody());
+    });
+
+    afterEach(() => {
+      jest.resetAllMocks();
+    });
+
+    it(`should return a ${HttpStatusCode.Ok} status code if the api request is successful`, async () => {
+      // Arrange
+      const mockedResponse = {
+        ...aGetFeeRecordCorrectionResponseBody(),
+        bankId,
+      };
+
+      jest.mocked(api.getFeeRecordCorrectionById).mockResolvedValue(mockedResponse);
+
+      // Act
+      await getFeeRecordCorrection(req, res);
+
+      // Assert
+      expect(res._getStatusCode()).toEqual(HttpStatusCode.Ok);
+    });
+
+    it(`should return the fee record correction in the response body if the api request is successful`, async () => {
+      // Arrange
+      const expectedResponse = {
+        ...aGetFeeRecordCorrectionResponseBody(),
+        bankId,
+      };
+
+      jest.mocked(api.getFeeRecordCorrectionById).mockResolvedValue(expectedResponse);
+
+      // Act
+      await getFeeRecordCorrection(req, res);
+
+      // Assert
+      expect(res._getData()).toEqual(expectedResponse);
+    });
+
+    it('should call the get fee record correction api endpoint once with the correct parameters', async () => {
+      // Act
+      await getFeeRecordCorrection(req, res);
+
+      // Assert
+      expect(api.getFeeRecordCorrectionById).toHaveBeenCalledTimes(1);
+      expect(api.getFeeRecordCorrectionById).toHaveBeenCalledWith(correctionId);
+    });
+
+    it(`should return a ${HttpStatusCode.NotFound} status code if the requesting users bank id does not match the bank id returned by the api`, async () => {
```

noticed the test for if fee record correction is not found has been removed and replaced by this.  Is it worth having both?

---

Good point but I think the case where a correction is not found (e.g., if removed) would be covered by the `should return a specific error code if an axios error is thrown` test case below (line 106, at present) as this would be a non-2xx code so would throw

As we're in `portal-api`, I think it's good to try and reduce the duplication of the controller tests in `central`, I've checked these and there's one for the not found case (`dtfs-central-api/src/v1/controllers/utilisation-report-service/fee-record-correction/get-fee-record-correction.controller/index.test.ts`)